v4l2/uapi/v4l2.py

```diff
@@ -3613,6 +3613,7 @@ class struct_v4l2_ext_control(Structure):
     'reserved2',
     'unnamed_1',
 ]
+struct_v4l2_ext_control._pack_ = 1
```

Comment:
This file is generated, you can't modify it as the modification will get overwritten. If this is really needed, then something needs to be done either in the generator script or __init__.py. Or maybe there's a bug in the ctypesgen, which needs to be fixed.

It's also not clear to me why this is needed or how it helps... struct_v4l2_ext_control has u32 fields, and the last one is an union. How does it get packed without this change? Which field gets extra alignment without this?

Reply:
Honestly I didn't really understand why this was needed...
To be honest, I used previously on my side another python library (pyv4l2 alternative) which were working correctly with ext ctrls ... I had to replicate this packing option to make pyv4l2 work correctly...
I would have been happy if you had a start of explanation :-D

Reply:
The `union_anon_15` is 8 bytes, and ctypes will align that to 8 bytes, which means 4 bytes of padding before the union. This increases the size of the struct from 20 to 24 bytes. The c-header is marked with __attribute__ ((packed)), so this looks like a bug in ctypesgen...

Reply:
https://github.com/ctypesgen/ctypesgen/issues/218